control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go

```diff
@@ -0,0 +1,208 @@
+package nto
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/metrics"
+	"github.com/openshift/hypershift/support/util"
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+const operatorName = "cluster-node-tuning-operator"
+
+type Images struct {
+	NodeTuningOperator string
+	NodeTunedContainer string
+}
+
+type Params struct {
+	ReleaseVersion          string
+	AvailabilityProberImage string
+	HostedClusterName       string
+	Images                  Images
+	OwnerRef                config.OwnerRef
+	DeploymentConfig        config.DeploymentConfig
+}
+
+// TODO use official NTO image
+func NewParams(hcp *hyperv1.HostedControlPlane, version string, images map[string]string, setDefaultSecurityContext bool) Params {
+	p := Params{
+		Images: Images{
+			NodeTuningOperator: images["cluster-node-tuning-operator"],
+			NodeTunedContainer: images["cluster-node-tuning-operator"],
+		},
+		ReleaseVersion: version,
+		OwnerRef:       config.OwnerRefFrom(hcp),
+	}
+
+	p.DeploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
+	p.DeploymentConfig.SetDefaults(hcp, nil, utilpointer.IntPtr(1))
+	p.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
+
+	p.DeploymentConfig.SetDefaultSecurityContext = setDefaultSecurityContext
+	p.HostedClusterName = hcp.Name
+
+	return p
+}
+
+func ReconcileClusterNodeTuningOperatorMetricsService(svc *corev1.Service, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(svc)
+
+	svc.Annotations = map[string]string{
+		"service.beta.openshift.io/serving-cert-secret-name": "node-tuning-operator-tls",
+	}
+
+	svc.Labels = map[string]string{"name": "node-tuning-operator", "hypershift.openshift.io/control-plane-component": "node-tuning-operator"}
+	svc.Spec.Ports = []corev1.ServicePort{
+		{
+			Port:       60000,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(60000),
+		},
+	}
+	svc.Spec.Selector = map[string]string{"name": operatorName}
+
+	return nil
+}
+
+func ReconcileClusterNodeTuningOperatorServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, ownerRef config.OwnerRef, clusterID string, metricsSet metrics.MetricsSet) error {
```

Comment:
There's no need for an ownerRef. The createOrUpdate function should now take care of that. We acually need to remove it from other reconcile funcs.

Reply:
Thanks, I removed these ownerRefs from the createOrUpdate functions here.